xapian-bindings/go/Makefile.am

```diff
@@ -0,0 +1,38 @@
+GOSRCDIR = ${GOROOT}/src/xapian
+TESTS = ./test
+AM_TESTS_ENVIRONMENT = \
+	abs_builddir='$(abs_builddir)' ;\
+	srcdir='$(srcdir)'
+	export abs_builddir ;\
+	export srcdir ;
+LOG_COMPILER = ${GO} test ./test
+
+install-data-hook:
+	cd ${GOROOT}/src/xapian
+	${GO} install -i -x -a
+
+all : modify
+	cd ${GOROOT}/src/xapian
+	${GO} build -i -x -a
+
+modify : build
+	mkdir -p ${GOSRCDIR}
+	cp -r * ${GOSRCDIR}
+
+build :
+	${SWIG} ${SWIG_FLAGS} -c++ -go -cgo -intgosize 64 go.i
+	awk 'NR==18{print "#cgo CXXFLAGS: ${XAPIAN_CXXFLAGS}"}1' xapian.go > xapian2.go
+	mv xapian2.go xapian.go
+	awk 'NR==19{print "#cgo CXXFLAGS: -std=gnu++11"}1' xapian.go > xapian2.go
+	mv xapian2.go xapian.go
+	awk 'NR==17{print "#cgo LDFLAGS: ${USE_LIB}"}1' xapian.go > xapian2.go
+	mv xapian2.go xapian.go
+	awk 'NR==17{print "#cgo CPPFLAGS: ${CPPFLAGS}"}1' xapian.go > xapian2.go
+	mv xapian2.go xapian.go
+	awk 'NR==17{print "#cgo LDFLAGS: -L${abs_top_srcdir}/../xapian-core/.libs -Wl,-rpath,${abs_top_srcdir}/../xapian-core/.libs -lxapian-1.5"}1' xapian.go > xapian2.go
```

Comment:
This is hard-coding information which should come from `XAPIAN_CXXFLAGS`.  Your hard-coded version will work currently when building from git, but will fail when building against an already installed xapian-core, or indeed once the current version is no longer 1.5.x.

Reply:
oh right....I didn't notice it..

Reply:
sir I have made changes for almost all that you have mentioned in the comments, for this one I need the path of shared library(libxapian-[verison.no].so) in the cloned git repo (source code) or if installed ..but the XAPIAN_CXXFLAGS is holding the include path for xapian lib and XAPIAN_LIBS is holding path for xapian lib tool archive (libxapian-1.5.la) ....so  since after running make, a (.libs) folder is created in xapian-core right..so im using path for it..and for -lxapian or -lxapian-1.5, there's a variable in xapian-core configure.ac called LIBRARY_VERSION_SUFFIX so can I use that ..? so that is could write it down as      -lxapian"$LIBRARY_VERSION_SUFFIX"            thank you :-)